libp2p/switch.nim

```diff
@@ -248,14 +253,19 @@ proc upgradeIncoming(s: Switch, conn: Connection) {.async, gcsafe.} =
 
     trace "Stopped secure handler", conn
 
-  if (await ms.select(conn)): # just handshake
-    # add the secure handlers
-    for k in s.secureManagers:
-      ms.addHandler(k.codec, securedHandler)
-
-  # handle un-secured connections
-  # we handshaked above, set this ms handler as active
-  await ms.handle(conn, active = true)
+  try:
+    if (await ms.select(incomingConn)): # just handshake
```

Comment:
this should really be refactored into a `handshake` named function (in a separate PR)

Reply:
Yep, but this is getting heavily reworked in a different PR and I want to scope this changes as much as possible.